pkg/client/unversioned/testclient/testclient.go

```diff
@@ -110,11 +113,27 @@ func (c *Fake) AddWatchReactor(resource string, reaction WatchReactionFunc) {
 	c.WatchReactionChain = append(c.WatchReactionChain, &SimpleWatchReactor{resource, reaction})
 }
 
+// PrependWatchReactor adds a reactor to the beginning of the chain
+func (c *Fake) PrependWatchReactor(resource string, reaction WatchReactionFunc) {
+	newChain := make([]WatchReactor, 0, len(c.WatchReactionChain)+1)
+	newChain = append(newChain, &SimpleWatchReactor{resource, reaction})
+	newChain = append(newChain, c.WatchReactionChain...)
+	c.WatchReactionChain = newChain
```

Comment:
This is kind of a long one-liner, but you could make a function so you don't have to repeat it all over.

```go
c.WatchReactionChain = append([]WatchReactor{&SimpleWatchReactor{resource, reaction}}, c.WatchReactionChain...)
```